pkg/stream/producer_daemon.go

```diff
@@ -175,11 +176,15 @@ func (d *producerDaemon) GetStage() int {
 }
 
 func (d *producerDaemon) Run(kernelCtx context.Context) error {
+	atomic.StoreInt32(&d.stopped, 0)
 	defer func() { atomic.StoreInt32(&d.stopped, 1) }()
 
 	// ensure we don't have a race with the code in Write checking if the ticker is nil
 	d.lck.Lock()
 	d.ticker = d.clock.NewTicker(d.settings.Interval)
+	if d.outCh.IsClosed() {
```

Comment:
not sure if this belongs here, but ideally we can get rid of it anyways because it isn't sound - we can get a call to write before our module starts, which would cause us to drop a message (because the output channel is still closed)